diff --git a/PKGBUILD b/PKGBUILD
index d1a0e3c..3bc0ba5 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -14,9 +14,11 @@ provides=('dwm')
 conflicts=('dwm')
 epoch=1
 source=(dwm.desktop
-        "$_pkgname::git+http://git.suckless.org/dwm")
+        "$_pkgname::git+http://git.suckless.org/dwm"
+        "dwm-config.def.h.diff")
 md5sums=('939f403a71b6e85261d09fc3412269ee'
-         'SKIP')
+         'SKIP'
+         '2df805b31d4ab7425fd4a9bc68a4ba86')

 pkgver(){
   cd $_pkgname
@@ -28,6 +30,23 @@ prepare() {
   if [[ -f "$SRCDEST/$pkgname/config.h" ]]; then
     cp -f "$SRCDEST/$pkgname/config.h" config.h
   fi
+  sed \
+    -e 's/CPPFLAGS =/CPPFLAGS +=/g' \
+    -e 's/CFLAGS =/CFLAGS +=/g' \
+    -e 's/LDFLAGS =/LDFLAGS +=/g' \
+    -e 's/_BSD_SOURCE/_DEFAULT_SOURCE/' \
+    -i config.mk
+  sed '/@tic/d' -i Makefile
+  for file in "${source[@]}"; do
+    if [[ "$file" == "config.h" ]]; then
+      # add config.h if present in source array
+      # Note: this supersedes the above sed to config.def.h
+      cp "$srcdir/$file" .
+    elif [[ "$file" == *.diff || "$file" == *.patch ]]; then
+      # add all patches present in source array
+      patch -Np1 <"$srcdir/$(basename ${file})"
+    fi
+  done
 }

 build() {
